### DISCUSSION TITLE
Getting content type error

### DISCUSSION THREAD

**Discussion by thexeromin** [actix/actix-web#3097] [Q&A] [answered]
Getting content type error in response when saving something in mongodb.
```rs
#[post("/")]
async fn add_link(client: web::Data<Client>, form: web::Form<Links>) -> impl Responder {
    let collection: Collection<Links> =
        client.database(DB_NAME).collection(COL_NAME);

    let new_doc = Links {
        id: None,
        ..form.into_inner()
    };

    let result = collection.insert_one(new_doc, None).await;

    match result {
        Ok(v) => {
            println!("{:#?}", v);
            HttpResponse::Ok().body("Link added")
        }
        Err(err) => HttpResponse::InternalServerError().body(err.to_string())
    }
}
```

**Comment by robjtede** [comment]
Can you include the error in the quesion, please?

**Reply by thexeromin** [reply]
When I am sending request to this route it's just showing response `Content type error `

**Reply by robjtede** [reply, marked as answer]
Using `form: web::Form<Links>` means the request needs to be of type `application/x-www-form-urlencoded` like that from a basic HTML form.

See docs: https://docs.rs/actix-web/latest/actix_web/web/struct.Form.html#extractor

**Reply by thexeromin** [reply]
Oh I see. I was sending json. Thanks
